File: C++/DM8/src/graphics/buffers/VertexArray.cpp
#include "VertexArray.h"


namespace DM8 {
	namespace graphics {

		VertexArray::VertexArray() 
		{
			glGenVertexArrays(1, &m_ID);
		}

		VertexArray::~VertexArray()
		{
			for (int i = 0; i < m_Buffers.size(); i++)
				delete m_Buffers[i];
		}

		void VertexArray::Bind() const
		{
			glBindVertexArray(m_ID);
		}

		void VertexArray::Unbind() const
		{
			glBindVertexArray(0);
		}


	}
}

File: Java/src/net/mirchs/dm8/graphics/buffers/VertexArray.java
package net.mirchs.dm8.graphics.buffers;

import static org.lwjgl.opengl.GL11.*;
import static org.lwjgl.opengl.GL15.*;
import static org.lwjgl.opengl.GL20.*;
import static org.lwjgl.opengl.GL30.*;

import java.util.ArrayList;
import java.util.List;

public class VertexArray {

	private int ID;
	private List<Buffer> buffers = new ArrayList<Buffer>();

	public VertexArray() {
		ID = glGenVertexArrays();
	}	
	
	public void addBuffer(Buffer buffer, int index) {
		bind();
		buffer.bind();
		
		glEnableVertexAttribArray(index);
		glVertexAttribPointer(index, buffer.getComponentCount(), GL_FLOAT, false, 0, 0);
		
		buffer.unbind();
		unbind();
		
		buffers.add(buffer);
	}

	public void bind() {
		glBindVertexArray(ID);
	}

	public void unbind() {
		glBindVertexArray(0);
	}


}
